engine/src/render/cascaded_shadow_maps_rasterizer/plugin.cpp

```diff
@@ -0,0 +1,314 @@
+using cubos::core::io::Window;
+using cubos::engine::RenderMeshVertex;
+
+namespace
+{
+    // Holds the data sent per scene to the GPU.
+    struct PerScene
+    {
+        glm::mat4 lightViewProj[cubos::engine::DirectionalShadowCaster::MaxSplits];
+        int numSplits;
+    };
+
+    // Holds the data sent per mesh to the GPU.
+    struct PerMesh
+    {
+        glm::mat4 model;
+    };
+
+    struct State
+    {
+        CUBOS_ANONYMOUS_REFLECT(State);
+
+        ShaderPipeline pipeline;
+        ShaderBindingPoint perSceneBP;
+        ShaderBindingPoint perMeshBP;
+
+        RasterState rasterState;
+        DepthStencilState depthStencilState;
+
+        VertexArray vertexArray;
+
+        ConstantBuffer perSceneCB;
+        ConstantBuffer perMeshCB;
+
+        State(RenderDevice& renderDevice, const ShaderPipeline& pipeline, VertexBuffer vertexBuffer)
+            : pipeline(pipeline)
+        {
+            perSceneBP = pipeline->getBindingPoint("PerScene");
+            perMeshBP = pipeline->getBindingPoint("PerMesh");
+            CUBOS_ASSERT(perSceneBP && perMeshBP, "PerScene and PerMesh binding points must exist");
+
+            rasterState = renderDevice.createRasterState({
+                .cullEnabled = true,
+                .cullFace = Face::Back,
+                .frontFace = Winding::CCW,
+            });
+
+            depthStencilState = renderDevice.createDepthStencilState({
+                .depth = {.enabled = true, .writeEnabled = true},
+            });
+
+            VertexArrayDesc desc{};
+            desc.elements[desc.elementCount++] = {
+                .name = "position",
+                .type = Type::UByte,
+                .size = 3,
+                .buffer = {.stride = sizeof(RenderMeshVertex),
+                           .offset = offsetof(RenderMeshVertex, position),
+                           .index = 0},
+            };
+            desc.buffers[0] = cubos::core::memory::move(vertexBuffer);
+            desc.shaderPipeline = pipeline;
+            vertexArray = renderDevice.createVertexArray(desc);
+
+            perSceneCB = renderDevice.createConstantBuffer(sizeof(PerScene), nullptr, Usage::Dynamic);
+            perMeshCB = renderDevice.createConstantBuffer(sizeof(PerMesh), nullptr, Usage::Dynamic);
+        }
+    };
+} // namespace
+
+/// @brief Gets a camera's frustum corners in world space.
+/// @param viewProj Matrix that transforms world space to the camera's clip space.
+/// @param corners Output vector where the corners will be stored.
+static void getCameraFrustumCorners(const glm::mat4& viewProj, std::vector<glm::vec4>& corners)
+{
+    for (int x = -1; x <= 1; x += 2)
+    {
+        for (int y = -1; y <= 1; y += 2)
+        {
+            for (int z = -1; z <= 1; z += 2)
+            {
+                auto corner = glm::inverse(viewProj) * glm::vec4(x, y, z, 1.0F);
+                corners.push_back(corner / corner.w);
+            }
+        }
+    }
+}
```

Comment:
This information will be useful in other regions of the code (i.e. frustum culling), maybe you can make it a static method of the new `Camera` component? Another option would be a global function in a `core/geom/frustum.hpp` or something (might make more sense in this case).